test/all.spec.ts

```diff
@@ -318,6 +334,95 @@ describe('validation', () => {
 		};
 		expect(f).to.throw();
 	});
+
+	it('should not fail when build config specifies valid network', async () => {
+		const f = () => {
+			compose.normalize({
+				version: '2.4',
+				networks: {
+					mynet: {},
+				},
+				services: {
+					main: {
+						build: { context: '.', network: 'mynet' },
+					},
+				},
+			});
+		};
+		expect(f).to.not.throw();
+	});
+
+	it('should fail when build config specifies invalid network', async () => {
+		const f = () => {
+			compose.normalize({
+				version: '2.4',
+				networks: {
+					othernet: {},
+				},
+				services: {
+					main: {
+						build: { context: '.', network: 'mynet' },
+					},
+				},
+			});
+		};
+		expect(f).to.throw();
+	});
+
+	it('should support extension fields', async () => {
+		const f = () => {
+			const c = compose.normalize({
+				version: '2.4',
+				services: {
+					main: {
+						build: '.',
+						'x-extension': true,
```

Comment:
I think something similar to this communicates intent a bit better:

```suggestion
						'x-my-custom-attribute': true,
```